libs/hdf-converters/src/conveyor-mapper.ts

```diff
@@ -0,0 +1,244 @@
+  const newout: Record<string, unknown>[] = [];
+  _.forEach(res, (value) => {
+    const temp = value as Record<string, unknown>;
+    const temp2: string = _.get(value, 'sha256') || '';
+    _.set(temp, 'filename', _.get(mapped, temp2));
+    const descriptions = _.map(
+      _.get(temp, 'result.sections') as Record<string, unknown>[],
+      (val) =>
+        createDescription(
+          val as Record<string, unknown>,
+          _.get(temp, 'result.score') as number,
+          _.get(temp, 'response.milestones.service_started') as string,
+          _.get(temp, 'response.service_name') as string
+        )
+    );
+    if (descriptions.length == 0) {
+      descriptions.push({
+        status: ExecJSON.ControlResultStatus.Passed,
+        code_desc: 'NA',
+        start_time: _.get(temp, 'response.milestones.service_started') as string
+      });
+    }
+    _.set(temp, 'result.sections', descriptions);
+    newout.push(temp);
+  });
+  const groups = _.groupBy(newout, (value) => {
+    return _.get(value, 'response.service_name');
+  });
+  return groups;
+}
+
+function controlMappingConveyor(): MappedTransform<
+  ExecJSON.Control & ILookupPath,
+  ILookupPath
+> {
+  return {
+    id: {path: 'sha256'},
+    title: {path: 'filename'},
+    desc: {path: 'cve_summary'},
+    impact: {
+      path: 'result.score',
+      transformer: (value) => {
+        return value / 1000;
+      }
+    },
+    refs: [],
+    tags: {
+      nist: DEFAULT_NIST_TAG
+    },
+    source_location: {
+      ref: ''
+    },
+    results: [
+      {
+        path: 'result.sections',
+        status: {path: 'status'},
+        code_desc: {path: 'code_desc'},
+        start_time: {path: 'start_time'}
+      }
+    ]
+  };
+}
+
+export class ConveyorMapper extends BaseConverter {
+  data: Record<string, unknown>;
+  type: string;
+  defaultMapping(): MappedTransform<ExecJSON.Execution, ILookupPath> {
+    return {
+      platform: {
+        name: 'Heimdall Tools',
+        release: HeimdallToolsVersion
+      },
+      version: HeimdallToolsVersion,
+      statistics: {},
+      profiles: [
+        {
+          name: this.type,
+          version: {path: 'api_server_version'},
+          title: {path: 'api_response.params.description'},
+          supports: [],
+          attributes: [],
+          groups: [],
+          status: 'loaded',
+          controls: [
+            {
+              path: 'api_response.results',
+              ...controlMappingConveyor()
+            }
+          ],
+          sha256: ''
+        }
+      ]
+    };
+  }
+  constructor(
+    conveyor: Record<string, unknown>,
+    meta: Record<string, unknown>,
+    type: string
+  ) {
+    const data = meta;
+    _.set(data, 'api_response.results', conveyor);
+    super(data);
+    this.type = type;
+    this.data = data;
+    this.setMappings(this.defaultMapping());
+  }
+}
+
+export class ConveyorResults {
+  data: unknown;
```

Comment:
i think this could be whatever the type is for the data coming in (maybe record<string, unknown>?)

Reply:
ask to see if they have a schema so that we don't need to do this cause we could just use types based off of their schema instead